From 5344ab962dcc6b1725b7bb3a8560b61ff3b8490d Mon Sep 17 00:00:00 2001
From: Pierre-Yves MORDRET <pierre-yves.mordret@st.com>
Date: Mon, 2 Dec 2019 15:26:16 +0100
Subject: [PATCH 2/2] clients: drop simple-dmabuf-drm

As stated by
https://gitlab.freedesktop.org/wayland/weston/merge_requests/234.
We drop this application as well as too tight to SoC and there are
too much driver-spefific code to allocate buffers.

Signed-off-by: Pierre-Yves MORDRET <pierre-yves.mordret@st.com>
---
 clients/meson.build | 43 -------------------------------------------
 meson_options.txt   |  8 --------
 2 files changed, 51 deletions(-)

diff --git a/clients/meson.build b/clients/meson.build
index ff6246d..5c56bb8 100644
--- a/clients/meson.build
+++ b/clients/meson.build
@@ -303,49 +303,6 @@ if get_option('demo-clients')
 	endforeach
 endif
 
-simple_dmabuf_drm_opts = get_option('simple-dmabuf-drm')
-simple_dmabuf_drm_deps = []
-foreach driver : [ 'etnaviv', 'intel', 'freedreno' ]
-	if simple_dmabuf_drm_opts.contains(driver)
-		required = true
-		enabled = true
-	elif simple_dmabuf_drm_opts.contains('auto')
-		required = get_option('auto_features').enabled()
-		enabled = not get_option('auto_features').disabled()
-	else
-		enabled = false
-	endif
-
-	if enabled
-		dep = dependency('libdrm_' + driver, required: false)
-		if dep.found()
-			simple_dmabuf_drm_deps += dep
-			config_h.set('HAVE_LIBDRM_' + driver.to_upper(), 1)
-		elif required
-			error('simple-dmabuf-drm is configured to use @0@ but it was not found. Or, you can remove @1@ from \'-Dsimple-dmabuf-drm\' list.'.format('libdrm_' + driver, driver))
-		endif
-	endif
-endforeach
-if simple_dmabuf_drm_deps.length() > 0
-	executable(
-		'weston-simple-dmabuf-drm',
-		'simple-dmabuf-drm.c',
-		xdg_shell_client_protocol_h,
-		xdg_shell_protocol_c,
-		fullscreen_shell_unstable_v1_client_protocol_h,
-		fullscreen_shell_unstable_v1_protocol_c,
-		linux_dmabuf_unstable_v1_client_protocol_h,
-		linux_dmabuf_unstable_v1_protocol_c,
-		include_directories: [ include_directories('..'), public_inc ],
-		dependencies: [
-			dep_wayland_client,
-			dep_libdrm,
-			simple_dmabuf_drm_deps
-		],
-		install: true
-	)
-endif
-
 if get_option('shell-desktop')
 	exe_keyboard = executable(
 		'weston-keyboard',
diff --git a/meson_options.txt b/meson_options.txt
index 254eb2b..3c2eb2f 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -179,13 +179,6 @@ option(
 	description: 'List of accessory clients to build and install'
 )
 option(
-	'simple-dmabuf-drm',
-	type: 'array',
-	choices: [ 'auto', 'intel', 'freedreno', 'etnaviv' ],
-	value: [ 'intel', 'freedreno', 'etnaviv' ],
-	description: 'List of DRM drivers to be supported by weston-simple-dmabuf-drm'
-)
-option(
 	'demo-clients',
 	type: 'boolean',
 	value: true,
@@ -198,7 +191,6 @@ option(
 	value: [ 'all' ],
 	description: 'Sample clients: simple test programs'
 )
-
 option(
 	'resize-pool',
 	type: 'boolean',
-- 
2.7.4

